components/admin_console/team_channel_settings/team/details/team_profile.jsx

```diff
@@ -29,6 +58,105 @@ export function TeamProfile({team, isArchived, isDisabled, onToggleArchive}) {
         archiveBtnID = t('admin.team_settings.team_details.archiveTeam');
         archiveBtnDefault = 'Archive Team';
     }
+
+    const toggleArchive = () => {
+        setOverrideRestoreDisabled(true);
+        onToggleArchive();
+    };
+    const button = () => {
+        if (restoreDisabled) {
```

Comment:
Or we may need to guard here given that it is currently showing the Upgrade option for self-managed servers